proto/substrait/algebra.proto

```diff
@@ -131,14 +131,42 @@ message ReadRel {
       message ArrowReadOptions {}
       message OrcReadOptions {}
       message DwrfReadOptions {}
+      message DelimeterSeparatedTextReadOptions {
+        // Delimiter separated files may be compressed.  The reader should
+        // autodetect this and decompress as needed.
+
+        // The character(s) used to separate fields.  Common values are comma,
+        // tab, and pipe.  Multiple characters are allowed.
+        string field_delimiter = 1;
+        // The maximum number of bytes to read from a single line.  If a line
+        // exceeds this limit the resulting behavior is undefined.
+        uint64 max_line_size = 2;
+        // The character(s) used to quote strings.  Common values are single
+        // and double quotation marks.
+        string quote = 3;
+        // If true, consume the first row as names of the columns.  These names
+        // are not used elsewhere in the plan.
+        uint64 header = 4;
+        // The character(s) used to escape characters in strings.  Backslash is
+        // the most common value.
+        string escape = 5;
+        // If this value is encountered (including empty string), the resulting
+        // value is null instead.  This value is checked before quotes are
```

Comment:
Why would this value be before quotes removed? That's super weird. The quotes are part of the container, not the value.

Reply:
If this is done after, you wouldn't be able to express both "NULL" and NULL.  By checking before you allow yourself the capability to find magic strings that wouldn't be present afterwards.

Reply:
Can you show me a set of systems that enable this behavior?

As an aside, I think that assumes a strange behavior for CSVs. My experience is that in CSV parsers, quotes are optional so "NULL" and NULL are fundamentally the same.

Reply:
In at least some flavors of CSV, quotes are not optional when fields contain special values; e.g. if the text contains a delimiter, the value must be quoted to prevent the character from understood as a delimiter. If your mental model of quoting is "treat this value literally" then I think it makes sense for NULL to mean NULL and "NULL" to mean "NULL". That said, while I think this is potentially useful behavior I also don't know of any systems which work this way. By contrast, I can immediately think of one that does have specialized NULL-handling for which "NULL" and NULL are treated identically -- R's read.csv.

Reply:
I think we're saying the same thing. 

I'm saying I'm not aware of any systems that will treat "NULL" and NULL differently in CSV reading. This is because many csv files only quote things when the values include delimiters. 

If systems don't treat them differently, I'm not clear why we're giving an option that forces us to treat them differently for null handling.

Reply:
Updated the documentation so the value is no longer evaluate unquoting.